exceptions/coordinator_result.hh

```diff
@@ -32,7 +32,8 @@ using coordinator_exception_container = utils::exception_container<
     mutation_write_timeout_exception,
     read_timeout_exception,
     read_failure_exception,
-    rate_limit_exception
+    rate_limit_exception,
+    overloaded_exception
```

Comment:
ok, since you add the exception here it can be justified to be a separate patch.

Reply:
I think it would still be a bit less confusing if `overloaded_exception` were introduced to the list of variants first, and only then new places which return it were introduced (instead of introducing `throw`s first and then changing them to returns).